vcd/resource_vcd_catalog_media.go

```diff
@@ -256,25 +258,27 @@ func genericVcdMediaRead(d *schema.ResourceData, meta interface{}, origin string
 	metadata, err := media.GetMetadata()
 	if err != nil {
 		log.Printf("[DEBUG] Unable to find media item metadata: %s", err)
-		return err
+		return diag.Errorf("%s", err)
```

Comment:
Not a disaster but you could use `diag` builting function which avoids unnecessary conversion throughout many places in this pr

```suggestion
		return diag.FromErr(err)
```

Reply:
good point, started to use but not everywhere yet, so made changes to the whole project.